### PR TITLE
Dedicated graph client for PubDev env

### PR DESCRIPTION
**Pull request by mingwli**
[AB#432334](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/432334)

### CODE REVIEW

File: src/docfx/config/Config.cs [hunk 1]
```diff
@@ -303,7 +303,9 @@ internal class Config : PreloadConfig
         /// <summary>
         /// Client id that can be used to access the Microsoft Graph API.
         /// </summary>
-        public string MicrosoftGraphClientId { get; init; } = "b6b77d19-e9de-4611-bc6c-4f44640ec6fd";
+        public string MicrosoftGraphClientId { get; init; } = EnvironmentVariable.DocsEnvironment == DocsEnvironment.Prod
```
Comment (yufeih) [inline src/docfx/config/Config.cs:306]: can this be set in `OpsXYZ`? in general try to put docs specific stuff there.

File: src/docfx/config/EnvironmentVariable.cs [hunk 2]
```diff
@@ -21,6 +21,11 @@ public static class EnvironmentVariable
 
         public static string? SessionId => GetValue("DOCFX_SESSION_ID");
 
+        public static readonly DocsEnvironment DocsEnvironment =
```
Comment (yufeih) [inline src/docfx/config/EnvironmentVariable.cs:24]: same here, try to limit `DOCS_` within `OpsXyz`

Comment (mingwli) [reply src/docfx/config/EnvironmentVariable.cs:24]: These can all be kept in `OpsAbc`, just one issue, how can `RegressionTest` use the `PubDev graph client`

Comment (yufeih) [reply src/docfx/config/EnvironmentVariable.cs:24]: regression test docfx config would override ops accessor

**Review by yufeih — APPROVED** [review]

**Pull request merged**